index.rst

```diff
@@ -42,20 +42,151 @@
+
+For a Pipeline to be executed the data ID needs to be converted to a quantum graph suitable for passing to the activator.
+The data ID would usually be the obsid for the data just acquired.
+Furthermore the group ID could be used to obtain all data that have previously been taken related to this observation.
 
 .. note::
 
-   **This technote is not yet published.**
+  No LSST rapid analysis pipelines are required to co-add data incrementally.
+  Pipelines should not co-add data as it arrives and should always defer co-addition until the final observation in the group has been taken.
+
+Single frame processing is the simplest mode whereby the products from the processing are not to be combined with subsequent observations.
+The more complex scenario relates to processing that requires co-addition where the co-addition can only occur at the end of data acquisition.
+
+One proposal for this is to have a chained pipeline approach.
+As data arrives they are processed with the nominated single frame processing variant.
+On completion they register themselves as inputs to a combination pipeline.
+This pipeline does not run automatically but waits to be told that all data have arrived.
+If an observation fails to reduce properly, or fails to pass some metrics threshold it will not register with the combination pipeline.
+
+When all expected data have arrived and been processed on their own, the final combination pipeline will be unblocked and will run to completion.
+Events will be issued on the completion of a single frame processing and on the completion of the combination pipeline.
+Metrics will also be published where appropriate during processing although the mechanism for this has not been defined.
+
+The issues that have to be solved are:
+
+1. How does the pipeline know to start processing a data ID?
+2. Which pipeline should be run for the given data ID?
+3. If the pipeline supports chaining, is that known to the system passing in the data?
+4. Who decides which observations should be combined and how?
+5. How does the system know the final observation arrived such that the combination pipeline can run?
+
+This document will describe two alternate scenarios for answering the above questions.
+The first describes a system entirely controlled by the observing script that is running the observation.
+The second describes a system that is decoupled from the observing system and is entirely driven by data arriving in the system.
+
+To simplify the remainder of this document the name given to the system that forwards processing requests to pipelines will be the "OCS Controlled Pipeline System (OCPS)."
+
+Observing Script Control
+========================
+
+In this scenario the observing script or notebook [*]_ has full control of all processing and the observing sequence and data processing are tightly coupled.
+The control flow could look something like this:
+
+* Tell OCPS that a new co-addition pipeline with the defined name is to be readied.
```

Comment:
Why does this need to be up front?

Reply:
That was from an initial discussion where people wanted to say in advance which pipelines they were going to be using and how many exposures to expect. You may be right that there is no gain to predeclaring the pipeline if it will work fine without knowing anything up front.

Reply:
Thinking further, a message such as this would be required in order to tell OCPS that any data it had got hanging around from previous observing should now be flushed.